packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -875,7 +875,58 @@ const schema: SchemaType<"Comments"> = {
     canRead: ['guests'],
     canUpdate: ['sunshineRegiment', 'admins'],
     canCreate: ['members', 'sunshineRegiment', 'admins'],
-  } 
+  },
+
+  doppelComments: {
+    // Resolved by doppelComments resolver
+    type: Array,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['admins'],
+    canUpdate: ['admins'],
+    hidden: true,
+  },
+
+  'doppelComments.$': {
+    type: Object,
+  },
+
+  ownDoppelCommentVote: resolverOnlyField({
+    type: Object,
+    graphQLtype: 'DoppelCommentVote',
+    canRead: ['guests'],
+    resolver: async (comment: DbComment, args: void, context: ResolverContext) => {
+      const { DoppelCommentVotes, currentUser } = context;
+      if (!currentUser) return null;
```

Comment:
I'd say we should probably feature-gate this but I expect it mostly end up using the `sqlResolver`, which I don't think currently provides an affordance for things like that.

Reply:
Provided the feature gate is statically defined and not read from a database setting you can do:
```
    sqlResolver: hasDoppelComments
      ? ({field, currentUserField}) => `(
        SELECT ROW_TO_JSON(dc.*)
        FROM "DoppelCommentVotes" dc
        WHERE dc."commentId" = ${field("_id")} AND dc."userId" = ${currentUserField("_id")}
      )`
      : undefined
```

 (I've not reviewed the rest of this PR - I just saw this comment)